internal/uvm/scsi.go

```diff
@@ -331,13 +350,144 @@ func (uvm *UtilityVM) GetScsiUvmPath(ctx context.Context, hostPath string) (stri
 }
 
 // grantAccess helper function to grant access to a file for the vm or vm group
-func (uvm *UtilityVM) grantAccess(ctx context.Context, hostPath string, vmAccess VMAccessType) error {
+func grantAccess(ctx context.Context, uvmID string, hostPath string, vmAccess VMAccessType) error {
 	switch vmAccess {
 	case VMAccessTypeGroup:
 		log.G(ctx).WithField("path", hostPath).Debug("granting vm group access")
 		return security.GrantVmGroupAccess(hostPath)
 	case VMAccessTypeIndividual:
-		return wclayer.GrantVmAccess(ctx, uvm.id, hostPath)
+		return wclayer.GrantVmAccess(ctx, uvmID, hostPath)
 	}
 	return nil
 }
+
+var _ = (Cloneable)(&SCSIMount{})
+
+// serializes the SCSIMount struct
+func (sm *SCSIMount) GobEncode() ([]byte, error) {
+	var buf bytes.Buffer
+	encoder := gob.NewEncoder(&buf)
+	errMsgFmt := "failed to encode SCSIMount: %s"
+	// encode only the fields that can be safely deserialized.
+	if err := encoder.Encode(sm.HostPath); err != nil {
+		return []byte{}, fmt.Errorf(errMsgFmt, err)
```

Comment:
Nit: Returning a nil slice makes more sense to me but it doesn't really matter. It's not being utilized in anyway if we detect an error is it?

Reply:
It won't be used if there is an error but I have changed it to return a nil slice.